models/Element/Service.php

```diff
@@ -742,13 +742,17 @@ public static function findForbiddenPaths($type, $user)
             $userCid[] = $userWorkspace['cid'];
         }
 
-        $roleWorkspacesSql = '
-            SELECT
+        $roleWorkspacesSql = 'SELECT
                 cpath, userid, max(list) as list
             FROM users_workspaces_'.$type.'
-            WHERE userId IN (' . implode(',', $userIds) . ') AND cid NOT IN ('.implode(',', $userCid).')
-            GROUP BY cpath
-        ';
+            WHERE userId IN (' . implode(',', $userIds) . ')';
+
+        if (!empty($userCid)) {
```

Comment:
The variable always exists. So this should be enough.
```suggestion
        if ($userCid) {
```

Reply:
I know what you mean, but I think this is slightly more readable as it is more explicit (and more in line with JavaScript and Python where `[]` is truthy).

Reply:
I would avoid empty when its possible, empty is slower than boolean cast.
https://www.contextualcode.com/Blog/php-micro-optimization.-variable-boolean-cast-vs-!empty
But it doesn't matter yes. Better avoid the sql if no role is set :)